execution-engine/contracts/hdac-system/pop/src/contract_votes.rs

```diff
@@ -0,0 +1,226 @@
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    string::String,
+};
+use core::fmt::Write;
+
+use contract::{contract_api::runtime, unwrap_or_revert::UnwrapOrRevert};
+use types::{
+    account::PublicKey,
+    system_contract_errors::pos::{Error, Result},
+    ApiError, Key, U512,
+};
+
+pub struct ContractVotes;
+pub struct Votes(pub BTreeMap<VoteKey, U512>);
+pub struct VoteStat(pub BTreeMap<PublicKey, U512>);
+
+#[derive(PartialOrd, Ord, PartialEq, Eq, Clone, Copy)]
+pub struct VoteKey {
+    user: PublicKey,
+    dapp: Key,
+}
+
+impl ContractVotes {
+    #[allow(clippy::or_fun_call)]
```

Comment:
It seems an unnecessary clippy suppression.

Reply:
https://github.com/hdac-io/CasperLabs/pull/14/commits/9464b5de93c7bca32387ba684abb64aee0814411